src/App.tsx

```diff
@@ -1,12 +1,60 @@
+/* eslint-disable react/no-unused-state */
 import React from 'react';
 import './App.scss';
 
-// import { getAll, get5First, getRed } from './api/goods';
-// or
-// import * as goodsAPI from './api/goods';
+import { getAll, get5First, getRedGoods } from './api/goods';
+import { GoodsList } from './components/GoodsList';
 
-const App: React.FC = () => (
-  <h1>Dynamic list of Goods</h1>
-);
+type State = {
+  goodsFromServer: Good[] | null
+};
+
+class App extends React.Component<{}, State> {
+  state: State = {
+    goodsFromServer: null,
+  };
+
+  render() {
+    return (
+      <div className="App">
+        <h1>Dynamic list of Goods</h1>
+        <button
+          type="button"
+          onClick={async () => {
+            this.setState({
+              goodsFromServer: await getAll(),
+            });
+          }}
```

Comment:
Create universal handler, and pass there a callback, for example: getGoods = `async (callback: () => Promise<Good[]>) {...}`